subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -1934,20 +1950,113 @@ static int do_write_op(struct lwm2m_engine_obj *obj,
 	}
 }
 
-static int get_observe_option(const struct zoap_packet *zpkt)
+static int get_option_int(const struct zoap_packet *zpkt, u8_t opt)
 {
 	struct zoap_option option = {};
 	u16_t count = 1;
 	int r;
 
-	r = zoap_find_options(zpkt, ZOAP_OPTION_OBSERVE, &option, count);
+	r = zoap_find_options(zpkt, opt, &option, count);
 	if (r <= 0) {
 		return -ENOENT;
 	}
 
 	return zoap_option_value_to_int(&option);
 }
 
+enum zoap_block_size lwm2m_default_block_size(void)
+{
+	switch (CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_COAP_BLOCK_SIZE) {
```

Comment:
You moved this block from a file that was only built when the option above was defined, so either protect this block or make this config generic.